quinn/src/connection.rs

```diff
@@ -54,31 +54,22 @@ impl Future for ConnectingFuture {
         if connected {
             let ConnectionDriver(conn) = self.0.take().unwrap();
             conn.lock().unwrap().driver.take();
-            Ok(Async::Ready(NewConnection::new(conn)))
+            Ok(Async::Ready(new_connection(conn)))
         } else {
             Ok(Async::NotReady)
         }
     }
 }
 
 /// A new connection (initiated locally or remotely)
-pub struct NewConnection {
-    /// The driver for the connection; this must be spawned for the connection to make progress
-    pub driver: ConnectionDriver,
-    /// The connection itself.
-    pub connection: Connection,
-    /// The stream of QUIC streams initiated by the client.
-    pub incoming: IncomingStreams,
-}
-
-impl NewConnection {
-    pub(crate) fn new(conn: ConnectionRef) -> Self {
-        Self {
-            driver: ConnectionDriver(conn.clone()),
-            connection: Connection(conn.clone()),
-            incoming: IncomingStreams(conn),
-        }
-    }
+pub type NewConnection = (ConnectionDriver, Connection, IncomingStreams);
```

Comment:
Would the generated documentation be clearer without this type alias?

Reply:
Yup. Fixed.